anomalib/loggers/wandb.py

```diff
@@ -0,0 +1,129 @@
+
+# Copyright (C) 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from typing import Any, List, Optional, Union
+
+import numpy as np
+import wandb
+from matplotlib.figure import Figure
+from pytorch_lightning.loggers.wandb import WandbLogger
+from pytorch_lightning.utilities import rank_zero_only
+
+from .base import ImageLoggerBase
+
+
+class AnomalibWandbLogger(ImageLoggerBase, WandbLogger):
+    """Logger for wandb.
+
+    Adds interface for `add_image` in the logger rather than calling the experiment object.
+    The rest is same as the WandbLogger provided by PyTorch Lightning and the doc string
+    for which is reproduced below:
+
+    Log using `Weights and Biases <https://www.wandb.com/>`_.
+
+    Install it with pip:
+
+    .. code-block:: bash
+
+        $ pip install wandb
+
+    Args:
+        name: Display name for the run.
+        save_dir: Path where data is saved (wandb dir by default).
+        offline: Run offline (data can be streamed later to wandb servers).
+        id: Sets the version, mainly used to resume a previous run.
+        version: Same as id.
+        anonymous: Enables or explicitly disables anonymous logging.
+        project: The name of the project to which this run will belong.
+        log_model: Save checkpoints in wandb dir to upload on W&B servers.
+        prefix: A string to put at the beginning of metric keys.
+        experiment: WandB experiment object. Automatically set when creating a run.
+        **kwargs: Arguments passed to :func:`wandb.init` like `entity`, `group`, `tags`, etc.
+
+    Raises:
+        ImportError:
+            If required WandB package is not installed on the device.
+        MisconfigurationException:
+            If both ``log_model`` and ``offline``is set to ``True``.
+
+    Example:
+        >>> from pytorch_lightning.loggers import WandbLogger
+        >>> from pytorch_lightning import Trainer
+        >>> wandb_logger = WandbLogger()
+        >>> trainer = Trainer(logger=wandb_logger)
+
+    Note: When logging manually through `wandb.log` or `trainer.logger.experiment.log`,
+    make sure to use `commit=False` so the logging step does not increase.
+
+    See Also:
+        - `Tutorial <https://colab.research.google.com/drive/16d1uctGaw2y9KhGBlINNTsWpmlXdJwRW?usp=sharing>`__
+          on how to use W&B with PyTorch Lightning
+        - `W&B Documentation <https://docs.wandb.ai/integrations/lightning>`__
+
+    """
```

Comment:
Is this from PL? For instance, I'm seeing this.

```
    Example:
        >>> from pytorch_lightning.loggers import WandbLogger
        >>> from pytorch_lightning import Trainer
        >>> wandb_logger = WandbLogger()
        >>> trainer = Trainer(logger=wandb_logger)
```

Reply:
My bad. I missed the example (it should be `AnomalibWandbLogger`) but rest is from pytorch lightning. As the only extensions are `save` and `add_image`. I've written it in the docstring above but maybe I can make it more prominent.